src/pm/hydra2/proxy/proxy.c

```diff
@@ -425,7 +425,7 @@ static HYD_status launch_processes(void)
 #define MAPPING_EXPECT_C(_c, _e) do { if (*(_c) != _e) MAPPING_PARSE_ERROR(); } while (0)
 #define MAPPING_EXPECT_AND_SKIP_C(_c, _e) do { MAPPING_EXPECT_C(_c, _e); ++c; } while (0)
 /* give an error iff the first |_m| characters of the string _s are equal to _e */
-#define MAPPING_EXPECT_S(_s, _e) (MPL_strncmp(_s, _e, strlen(_e)) == 0 && !MAPPING_ISIDENT((_s)[strlen(_e)]))
+#define MAPPING_EXPECT_S(_s, _e) (strncmp(_s, _e, strlen(_e)) == 0 && !MAPPING_ISIDENT((_s)[strlen(_e)]))
```

Comment:
An unrelated comment: if we are calling `strlen` then there really isn't need to use `strncmp`, because if the string is not terminated, `strlen` will have as much trouble as `strcmp` in the first place, so `strncmp` doesn't matter.

Reply:
True. I actually ended up looking at this because of your `.local` comparison PR and came to the same conclusion that `strlen` is going to do what it does, so `strncmp` won't be of any help :). Then I went looking for `strncmp` and was surprised to see this MPL version.